File: java/bob/src/main/java/Bob.java
class Bob {

    String hey(String input) {

        String inputString = input.trim();

        if (inputString.isEmpty()) {
            return "Fine. Be that way!";
        }

        if (isYelling(inputString) && isQuestion(inputString)) {
            return "Calm down, I know what I'm doing!";
        }

        if (isQuestion(inputString)) {
            return "Sure.";
        }

        if (isYelling(inputString)) {
            return "Whoa, chill out!";
        }

        return "Whatever.";
    }

    private boolean isQuestion(String input) {
        return input.matches(".*\\?$");
    }

    private boolean isYelling(String input) {
        String str = input.replaceAll("[^a-zA-Z]", "");
        return !str.isEmpty()
                && str.chars().mapToObj(c -> (char) c).allMatch(Character::isUpperCase);
    }

}

File: javascript/bob/bob.js
export const hey = (message) => {

    message = message.trim();

    if (message === "") {
        return "Fine. Be that way!";
    }

    if (isQuestion(message) && isYelling(message)) {
        return "Calm down, I know what I'm doing!";
    }

    if (isYelling(message)) {
        return "Whoa, chill out!";
    }

    if (isQuestion(message)) {
        return "Sure.";
    }

    
    return "Whatever.";
};

const isQuestion = (message) => /.*\?$/g.test(message);
const isYelling = (message) => {
    let r = message.replaceAll(/[^a-zA-Z]/g, '');
    return /^[A-Z]+$/g.test(r);
}
